packages/@lwc/integration-karma/test/swapping/components/index.spec.js

```diff
@@ -42,5 +44,15 @@ if (process.env.NODE_ENV !== 'production') {
                 swapComponent(D, function () {});
             }).toThrow();
         });
+
+        it('should be a no-op for non components', () => {
+            const elm = createElement('x-container', { is: Container });
+            document.body.appendChild(elm);
+            expect(elm.testValue).toBe('I may look like a component');
+            expect(swapComponent(Z, X)).toBe(false);
+            return Promise.resolve().then(() => {
+                expect(elm.testValue).toBe('I may look like a component');
+            });
+        });
```

Comment:
How about tests to assert the `TypeError`s above? I.e. Z is a component and X is not, and vice versa.

Reply:
There were existing tests for those cases, I updated them to check for specific message that I introduced.